### ISSUE TITLE
[Bug]: Stretchly on M1 seems to be consuming too much energy

### ISSUE THREAD

**Issue by acyment** [hovancik/stretchly#1278]
### Version

- [X] I'm using version 1.13.1

### Known issues

- [X] I've checked Known issues

### Existing issues

- [X] I've checked Existing issues

### What operating system are you using?

macOS

### Operating System Version

Ventura 13.2.1

### Reproduction steps

1. Open Stretchly
2. Open Activity Monitor
3. Check energy consumption

### Expected Behavior

I would expect a fairly passive app like Stretchly to be in the low consuming spectrum. I am currently using Time Out, not because I like the experience better, but because consumption seems to be approximately 5x lower.

Thanks for the hard work!

### Actual Behavior

<img width="798" alt="Screenshot 2023-03-03 at 10 52 02" src="https://user-images.githubusercontent.com/1872616/222737814-88314ff1-8b3b-4b22-b530-fea81ce17c1b.png">

<img width="953" alt="Screenshot 2023-03-03 at 10 52 18" src="https://user-images.githubusercontent.com/1872616/222737865-04828d66-e890-4bb6-b619-e64e17faab4a.png">

Apple's algorithm for estimating energy consumption is closed source (sucks), but there seems to be a correlation with the amount of threads and idle wake ups. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hovancik** [comment]
Hi,I don't have mac with M, but Stretchly is build with ElectronJS which is resource hungry ;/ 

`there seems to be a correlation with the amount of threads and idle wake ups.` Do you mean it goes up when your computer comes back from idle? Number of processes that Stretchly has should stay same-ish over time. (Processes get created when break starts but they should get destroyed when they end)

**Comment by 12people** [comment]
@hovancik 
Just wondering — have you considered replacing Electron with [Tauri](https://tauri.app/)? Just like Electron, it's a framework for building desktop apps with web technologies, but unlike Electron, it uses a platform's native WebView rather than bundling Chromium and is more optimized for a smaller footprint and lower memory usage.

That could perhaps help make Stretchly more efficient on all platforms.

**Comment by hovancik** [comment]
Yes, I did when Stretchly was not so big. At that time Electron alternatives didn't have Tray or Building packages.

Some of them have now, but with the current amount of code it basically means starting from scratch.

**Comment by 12people** [comment]
Oh, ok, I understand.

**Comment by StevenM1** [comment]
I'm on an M1 Pro here, macOS Monterey (12.7.1) and stretchly 1.14.1, but I don't have the same excessive amount of resources used as reported here.. It's still an electron app, of course, so its fingerprint is not as small as a native app could get to, but it seems reasonable? Reported 12h power is 0.55 (whatever that means) here.
<img width="1818" alt="Screenshot 2023-11-06 at 13 00 27" src="https://github.com/hovancik/stretchly/assets/17780626/81421aff-3c72-4e62-a772-feac419602a5">

**Comment by mundeli** [comment]
***FIrst, thank you so much for this app!***
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;My body and brain thank you too!

I also noticed power consumption on Mac.  Have an M2 chip

<img width="186" height="160" alt="Image" src="https://github.com/user-attachments/assets/c1e9ac5e-e30d-4684-aacb-0471c5823587" />

Went away for a day, turned off wifi, and put computer into sleep mode, disconnected power (thunderstorms predicted)
Came back and battery was drained.
Stretchy was one of the power hungry.

<img width="765" height="273" alt="Image" src="https://github.com/user-attachments/assets/ef66b51d-39a5-4a4f-b059-709546fe7ab8" />

Cheers from the US,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Open source power!

**Comment by hovancik** [comment]
@mundeli Stretchly is built on Electron (which bundles whole Chrome/Node) - similar to Teams. Not sure if we can do much here, unless we rewrite to something else.

**Comment by hovancik** [comment]
I'll try to make some improvements.

**Comment by mundeli** [comment]
Off development topic:  
I've had back surgery, and I'm a coder, as such I can disappear into the code and neglect proper care of my back.
Your Stretchly tool prompts me to be better to myself, now, and I'll reap rewards into the future.

--
And, as we all have experienced:  sometimes the solution to a coding challenge becomes manifest on return to the keyboard following a break.

So again, thank you for your work and dedication!
Bruce

**Comment by hovancik** [comment]
Thanks! I love hearing stories how Stretchly helps :) Best of luck!

**Comment by hovancik** [comment]
Added some improvements

**Comment by mundeli** [comment]
I'll update, and monitor, and get back to you.
Thank you,
Bruce

**Comment by hovancik** [comment]
Thanks, please not that new release will take some time... there are still PRs I need to look at
